server/endpoint/lister/endpoint.go

```diff
@@ -28,16 +27,14 @@ const (
 
 // Config defines which configuration our endpoint expects.
 type Config struct {
-	Logger     micrologger.Logger
-	Middleware *middleware.Middleware // nolint: structcheck, unused
-	Service    *lister.Service
+	Logger  micrologger.Logger
+	Service *lister.Service
 }
 
 // Endpoint is the actual endpoint data structure.
 type Endpoint struct {
-	logger     micrologger.Logger
-	middleware *middleware.Middleware // nolint: structcheck, unused
-	service    *lister.Service
+	logger  micrologger.Logger
+	service *lister.Service
```

Comment:
I don't _think_ removing middleware would be an issue, but I'd want to be extremely confident before doing so. Given that this project will be completely dead soon, I would recommend suppressing the linter finding for the unused middleware field instead of removing it at this point, unless you're very sure it's safe (this also means external tools that import this library don't expect to have a Middleware field).